pkg/controller/apischeme/apischeme_controller_test.go

```diff
@@ -0,0 +1,415 @@
+package apischeme
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	configv1 "github.com/openshift/api/config/v1"
+	cloudingressv1alpha1 "github.com/openshift/cloud-ingress-operator/pkg/apis/cloudingress/v1alpha1"
+
+	awsclient "github.com/openshift/cloud-ingress-operator/pkg/awsclient/mock"
+
+	utils "github.com/openshift/cloud-ingress-operator/pkg/controller/utils"
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const masterMachineLabel string = "machine.openshift.io/cluster-api-machine-role"
+const defaultRegionName string = "us-east-1"
+const defaultAzName string = "us-east-1a"
+const defaultAPIEndpoint string = "https://api.unit.test:6443"
+const defaultClusterDomain string = "unit.test"
+
+type mocks struct {
+	fakeKubeClient client.Client
+	mockCtrl       *gomock.Controller
+	mockAws        *awsclient.MockClient
+	scheme         *runtime.Scheme
+}
+
+// setupTests will make sure each test has the same starting point
+func setupTests(t *testing.T, localObjs []runtime.Object) *mocks {
```

Comment:
this is a great way to refactor out all the schemes from each tests. I will borrow this idea on my next unit tests :)